contract-tools/xchain/xchain.go

```diff
@@ -666,37 +721,79 @@ func (a *aggregator) sendDeal(ctx context.Context, aggCommp cid.Cid, transferID
 		return fmt.Errorf("failed to get chain ID: %w", err)
 	}
 	// Encode the chainID as uint256
-	encodedChainID, err := encodeChainID(chainID)
+	// intEncodedChainID, err := encodeChainID(chainID)
+	// if err != nil {
+	// 	return fmt.Errorf("failed to encode chainID: %w", err)
+	// }
+
+	clientAddr, err := a.lotusAPI.WalletDefaultAddress(ctx)
+	if err != nil {
+		return err
+	}
+	dummyBuf, err := cborutil.Dump("dummy")
+	if err != nil {
+		return err
+	}
+	dummySig, err := a.lotusAPI.WalletSign(ctx, clientAddr, dummyBuf)
+	if err != nil {
+		return fmt.Errorf("wallet sign failed: %w", err)
+	}
+	b2sum := blake2b.Sum256(dummyBuf)
+	pubk, err := gocrypto.EcRecover(b2sum[:], dummySig.Data)
 	if err != nil {
-		return fmt.Errorf("failed to encode chainID: %w", err)
+		return err
 	}
-	dealLabel, err := market.NewLabelFromBytes(encodedChainID)
+	pubkHash := ethcrypto.Keccak256(pubk[1:]) // Skip the first byte (0x04) which indicates uncompressed public key
+	ethAddress := common.BytesToAddress(pubkHash[12:]) // Take the last 20 bytes
+	encodedLabel, err := encodeChainIDAndAddress(chainID, ethAddress)
+	if err != nil {
+		return fmt.Errorf("failed to encode chainID and address: %w", err)
+	}
+	labelString := hex.EncodeToString(encodedLabel)
+
+	dealLabel, err := market.NewLabelFromString(labelString)
```

Comment:
In the same way I think the changes to the Prover-Axelar contract should be a separate contract I think this functionality should be behind a config option.  xchain code here should support both ways of doing deal.  

If this integration is too annoying I think we should keep this change in a different branch or in a fork.  I don't want the basic prototype flow to be lost with the latest evolution.